src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -1510,6 +1520,198 @@ class spell_dru_wild_growth : public SpellScriptLoader
+                else if (spellInfo->Id == wrath)
+                    OnSpellCast(SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_LUNAR_AURA);
+            }
+        }
+    }
+
+    void HandleApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+    {
+        if (Unit* owner = GetUnitOwner())
+        {
+            spell_dru_eclipse_common::SetSpellCount(owner, SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, aurEff->GetAmount());
+            spell_dru_eclipse_common::SetSpellCount(owner, SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, aurEff->GetAmount());
+        }
+    }
+
+    void HandleRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Unit* owner = GetUnitOwner())
+        {
+            owner->RemoveAura(SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT);
+            owner->RemoveAura(SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT);
+        }
+    }
+    
+    void OnOwnerOutOfCombat()
+    {
+        if (Unit* owner = GetUnitOwner())
+            owner->AddAura(SPELL_DRUID_ECLIPSE_OCC, owner);
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_dru_eclipse_dummy::HandleApply, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        OnEffectRemove += AuraEffectApplyFn(spell_dru_eclipse_dummy::HandleRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        OnProc += AuraProcFn(spell_dru_eclipse_dummy::HandleProc);
+        OnOutOfCombat += OutOfCombatFn(spell_dru_eclipse_dummy::OnOwnerOutOfCombat);
+    }
+
+private:
+    void OnSpellCast(uint32 cntSpellId, uint32 otherCntSpellId, uint32 eclipseAuraSpellId)
+    {
+        if (Unit* owner = GetUnitOwner())
+        {
+            if (Aura* aura = owner->GetAura(cntSpellId))
+            {
+                uint32 remaining = aura->GetStackAmount();
+                if (remaining == 0)
+                    return;
+
+                if (remaining > 1)
+                    aura->SetStackAmount(remaining - 1);
+                else
+                {
+                    // cast eclipse
+                    owner->CastSpell(owner, eclipseAuraSpellId, TRIGGERED_FULL_MASK);
+
+                    // Remove stacks from other one as well
+                    // reset remaining power on other spellId
+                    owner->RemoveAura(cntSpellId);
+                    owner->RemoveAura(otherCntSpellId);
+                }
+            }
+        }
+    }
+};
+
+// 48517 Eclipse (Solar) + 48518 Eclipse (Lunar)
+class spell_dru_eclipse_aura : public AuraScript
+{
+    PrepareAuraScript(spell_dru_eclipse_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_ECLIPSE_LUNAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_SOLAR_SPELL_CNT, SPELL_DRUID_ECLIPSE_DUMMY });
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
```

Comment:
owner / nullcheck

always keep in mind: caster may be null in AuraScripts. Imagine i'm dotting you with moonfire and logout after it => caster null.